src/app/patient-dashboard/common/drug-orders/drug-order-set/drug-order-set.component.ts

```diff
@@ -0,0 +1,302 @@
+    this.subscription = this.patientService.currentlyLoadedPatient.subscribe(
+      (patient) => {
+        if (patient) {
+          this.patient = patient;
+          const amrsId = _.find(this.patient.identifiers.openmrsModel,
+            (identifer: any) => {
+              if (identifer.identifierType.uuid === '58a4732e-135x9-11df-a1f1-0026b9348838') {
+                return true;
+              }
+            });
+          if (amrsId) {
+            this.patientIdentifer = amrsId.identifier;
+          }
+          this.getDrugOrdersSet();
+        }
+      }
+    );
+  }
+
+  public getProvider() {
+    this.drugOrderService.getProviderByPersonUuid(this.personUuid).subscribe((data) => {
+      this.provider = data.providerUuid;
+    });
+  }
+  public getDrugOrdersSet() {
+    this.fetchingResults = true;
+    const orderSets = [];
+    this.isBusy = true;
+    const patientUuId = this.patient.uuid;
+    this.orderSetResourceService.getAllOrderSets()
+      .subscribe((data) => {
+        data = data.results;
+        data.forEach((value) => {
+          orderSets.push(value);
+        });
+        if (orderSets) {
+          this.drugOrderSets = orderSets.reverse();
+          console.dir(this.drugOrderSets);
+          this.selectedOrderSets = this.drugOrderSets;
+          console.log(this.selectedOrderSets, 'selected');
+          this.orderSetTemplateConverter();
+          this.drugOrderSets.forEach((value) => {
+            if (value.dateStopped) {
+              this.inactiveDrugOrderSets.push(value);
+            } else if (!value.dateStopped) {
+              this.activeDrugOrderSets.push(value);
+            }
+          });
+          this.fetchingResults = false;
+          this.loadingDrugOrderSetStatus = false;
+          this.isBusy = false;
+        }
+      });
+  }
+  public orderSetTemplateConverter() {
+
+    for (let i = 0; i < this.selectedOrderSets.length; i++) {
+      const orderTemplate = this.selectedOrderSets[i].orderSetMembers;
+
+      for (let b = 0; b < orderTemplate.length; b++) {
+        this.selectedOrderSets[i].concept = orderTemplate[b].concept;
+        const jsonTemplate = JSON.parse(orderTemplate[b].orderTemplate);
+        this.selectedOrderSets[i].template = jsonTemplate;
+      }
+      setTimeout(() => {
+        this.filterOrders(this.selectedOrderSets);
+      }, 200);
+
+    }
+    console.log('orderset', this.selectedOrderSets);
+
+  }
+  private filterOrders(orderSets) {
+    const orderSetStatus = [];
+    orderSets.forEach((value) => {
+      if (value.dateStopped) {
+        orderSetStatus.push('INACTIVE');
+      } else if (!value.dateStopped) {
+        orderSetStatus.push('ACTIVE');
+      }
+    });
+    if (orderSetStatus.length > 0) {
+      this.orderStatus = this.getUniqueNames(orderSetStatus);
+    }
+  }
+  private getUniqueNames(originArr) {
+    const newArr = [];
+    const originLength = originArr.length;
+    let found, x, y;
+    for (x = 0; x < originLength; x++) {
```

Comment:
I would prefer using _.each instead of native forLoop. I am not aware of any performance glitches though.